pkg/providers/labels/labels_test.go

```diff
@@ -0,0 +1,235 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package labels_test
+
+import (
+	"testing"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1beta1"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/labels"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+func TestGetAllSingleValuedRequirementLabels(t *testing.T) {
+	cases := []struct {
+		name           string
+		requirements   scheduling.Requirements
+		expectedLabels map[string]string
+	}{
+		{
+			name:           "Nil instance type",
+			requirements:   nil,
+			expectedLabels: map[string]string{},
+		},
+		{
+			name: "Single-valued requirements",
+			requirements: scheduling.NewRequirements(
+				scheduling.NewRequirement("node.kubernetes.io/instance-type", corev1.NodeSelectorOpIn, "Standard_D2s_v3"),
```

Comment:
```suggestion
				scheduling.NewRequirement(corev1.LabelInstanceTypeStable, corev1.NodeSelectorOpIn, "Standard_D2s_v3"),
```
(here and elsewhere)